app/src/main/java/com/github/steroidteam/todolist/view/adapter/TodoAdapter.java

```diff
@@ -37,6 +51,46 @@ public void onBindViewHolder(@NonNull TaskHolder holder, int position) {
         Task currentTask = todoList.getTask(position);
 
         holder.taskBody.setText(currentTask.getBody());
+        if (currentTask.getDueDate() != null) {
+            Context context = holder.itemView.getContext();
+            int lightGrey = ContextCompat.getColor(context, R.color.light_grey);
+            // The ImageSpan works by replacing a substring by the image (in this case, the clock
+            // icon). We add two leading spaces here to:
+            //  1) Keep useful text from being overwritten by the image.
+            //  2) Add a little margin between the icon and the date.
+            SpannableString dueDate =
+                    new SpannableString("  " + formatDate(currentTask.getDueDate()));
+            // Make the due date 75% the size of the task's body.
+            dueDate.setSpan(
+                    new RelativeSizeSpan(0.75f),
+                    0,
+                    dueDate.length(),
+                    Spanned.SPAN_EXCLUSIVE_INCLUSIVE);
+            // Tint the clock icon with the same colour as the text (light grey).
+            Drawable clockIcon =
+                    DrawableCompat.wrap(
+                            Objects.requireNonNull(
+                                    ContextCompat.getDrawable(context, R.drawable.clock_icon)));
+            DrawableCompat.setTint(clockIcon, lightGrey);
+            clockIcon.setBounds(
+                    0, 0, clockIcon.getIntrinsicWidth(), clockIcon.getIntrinsicHeight());
+            // Add the clock icon to the start of the span.
+            dueDate.setSpan(
+                    new ImageSpan(clockIcon, DynamicDrawableSpan.ALIGN_BASELINE),
+                    0,
+                    1,
+                    Spanned.SPAN_EXCLUSIVE_INCLUSIVE);
+            // Make the text's colour light grey as well.
+            dueDate.setSpan(
+                    new ForegroundColorSpan(lightGrey),
+                    0,
+                    dueDate.length(),
+                    Spanned.SPAN_EXCLUSIVE_INCLUSIVE);
+            // Add some margin between the task's body and the due date.
+            holder.taskBody.append("   ");
+            // Insert the span with the due date.
+            holder.taskBody.append(dueDate);
```

Comment:
Maybe you can consider having a private method for this.

Reply:
Great idea!